src/pages/VenuePage/VenuePage.tsx

```diff
@@ -38,25 +42,105 @@ import { useVenueId } from "hooks/useVenueId";
-import { LoadingPage } from "components/molecules/LoadingPage/LoadingPage";
-// import { AccessDeniedModal } from "components/atoms/AccessDeniedModal/AccessDeniedModal";
-import TemplateWrapper from "./TemplateWrapper";
-
 import { updateTheme } from "./helpers";
 
+import Login from "pages/Account/Login";
+
 import "./VenuePage.scss";
 
-import Login from "pages/Account/Login";
-import { useRelatedVenues } from "hooks/useRelatedVenues";
+const CountDown = React.lazy(() =>
+  import("components/molecules/CountDown").then((m) => ({
+    default: m.CountDown,
+  }))
+);
+const LoadingPage = React.lazy(() =>
+  import("components/molecules/LoadingPage/LoadingPage").then((m) => ({
+    default: m.LoadingPage,
+  }))
+);
+const TemplateWrapper = React.lazy(() => import("./TemplateWrapper"));
 
 // @debt Refactor this constant into settings, or types/templates, or similar?
 const hasPaidEvents = (template: VenueTemplate) => {
   return template === VenueTemplate.jazzbar;
 };
 
+interface PreloadProps {
+  venue?: AnyVenue;
+}
+
+const Preload: React.FC<PreloadProps> = ({ venue }) => (
+  <Helmet>
+    {
+      // preload the icon only when TemplateWrapper with the navbar icon aren't loaded lazily
+      // <link href={SPARKLE_ICON} rel="preload" as="image" />
+    }
+    {venue?.mapBackgroundImageUrl && (
+      <link href={venue?.mapBackgroundImageUrl} rel="preload" as="image" />
+    )}
+    {
+      // NOTE: if there is significant number of rooms, preload might degrade performance, cut down to few
+      (venue?.rooms ?? []).map(
+        (room) =>
+          room?.image_url && (
+            <link
+              key={room?.image_url}
+              href={room.image_url}
+              rel="preload"
+              as="image"
+            />
+          )
+      )
+    }
+  </Helmet>
+);
+
+const usePreloaded = (venueId?: string) => {
+  const { loading, error, value } = useAsync(async () => {
+    if (!venueId) return;
+
+    return tracePromise(
+      "usePreloadedVenue::getVenue",
+      () =>
+        firebase
+          .functions()
+          .httpsCallable("venue-getVenue")({ venueId })
+          .then((result) => {
+            const data: {
+              venue?: WithId<AnyVenue>;
+              events?: WithId<VenueEvent>[];
+            } = result.data;
+            return {
+              venue: data?.venue,
+              events: isDefined(data?.events)
+                ? data.events.map((event) => withVenueId(event, venueId))
+                : undefined,
+            };
+          }),
+      {
+        attributes: { venueId },
+        withDebugLog: true,
+      }
+    );
+  }, [venueId]);
+
+  return {
+    venue: value?.venue,
+    events: value?.events,
+    error,
+    isLoaded: !loading && !error,
+  };
+};
+
 export const VenuePage: React.FC = () => {
   const venueId = useVenueId();
+  const {
+    venue: preVenue,
+    events: preEvents,
+    error: preError,
+    isLoaded: preLoaded,
+  } = usePreloaded(venueId);
+
```

Comment:
If this is going to be kept, then it needs to be moved out of this component file to somewhere in `hooks/*`.

As mentioned in the backend function file itself, adding events in here defeats the purpose that this function was originally created for (to get the venue config in the quickest and most minimal way possible). I don't believe we have any need to fetch the events this early. I also believe we don't want to be making a function that exposes events to unauthed users for security/privacy reasons.

After the 'slimming down' of our standard data fetching methods, do we even need this at all anymore? A fallback 'in case' isn't a good reason to take on extra code complexity/etc.

Reply:
Removed entire logic and replaced with hook (`src/hooks/usePreloadAssets.ts`) that doesn't use Helmet library in e9da1d5788e915f6de13a7aa0315601017c75b73

Reply:
Removed extra requests by removing the BE function in 0da80c02d45c99456fdaff937409dc563ca3fdb4
No more extra fallback data.